src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/ProvideSemanticTokensResponse.cs

```diff
@@ -1,21 +1,41 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 #pragma warning disable CS0618
+using System;
 using OmniSharp.Extensions.LanguageServer.Protocol.Models.Proposals;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
 {
     public class ProvideSemanticTokensResponse
     {
-        public SemanticTokens Result { get; private set; }
+        public SemanticTokens Result { get; }
 
-        public long? HostDocumentSyncVersion { get; private set; }
+        public long? HostDocumentSyncVersion { get; }
 
         public ProvideSemanticTokensResponse(SemanticTokens result, long? hostDocumentSyncVersion)
         {
             Result = result;
             HostDocumentSyncVersion = hostDocumentSyncVersion;
         }
+
+        public override bool Equals(object obj)
+        {
+            if(obj is ProvideSemanticTokensResponse)
+            {
+                var other = obj as ProvideSemanticTokensResponse;
+
+                return other.HostDocumentSyncVersion.Equals(HostDocumentSyncVersion) && other.Result.Equals(Result);
+            }
+            else
+            {
+                return false;
+            }
+        }
```

Comment:
```suggestion
        public override bool Equals(object obj) => 
            obj is ProvideSemanticTokensResponse other &&
            other.HostDocumentSyncVersion.Equals(HostDocumentSyncVersion) && 
            other.Result.Equals(Result);
```

Reply:
Yeah. When do we pick up the SDK with records in it? My PREFERED fix here is that I just make the class a record and be done with it.